### PR TITLE
[fix](memory) Refactor LRU cache policy memory tracking

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Fix #35590, CacheManager get CachePolicy should be used lock.
- Add LRUCachePolicyTrackingAllocator and LRUCachePolicyTrackingManual.
- tracking memory in LRUCachePolicy::insert, LRUCacheValueBase not need init `mem_tracker`, so not need to get CachePolicy under lock.

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Review by jacktengg — COMMENTED** [review]
Should not change apache-orc submodule and be/src/clucene

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/runtime/memory/lru_cache_policy.h [hunk 1]
```diff
@@ -208,6 +195,116 @@ class LRUCachePolicy : public CachePolicy {
...
+                                    bool cache_value_check_timestamp, bool enable_prune = true)
+            : LRUCachePolicy(type, capacity, lru_cache_type, stale_sweep_time_s, num_shards,
+                             element_count_capacity, cache_value_time_extractor,
+                             cache_value_check_timestamp, enable_prune) {
+        init_mem_tracker(lru_cache_type_string(lru_cache_type));
+    }
+
+    ~LRUCachePolicyTrackingAllocator() override { reset_cache(); }
+
+    void init_mem_tracker(const std::string& type_name) {
```
Comment (HappenLee) [inline be/src/runtime/memory/lru_cache_policy.h:230]: should be private `_init_mem_tracker`

Comment (xinyiZzz) [reply be/src/runtime/memory/lru_cache_policy.h]: done

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**